models/git/branch.go

```diff
@@ -139,6 +141,16 @@ func (b *Branch) LoadPusher(ctx context.Context) (err error) {
 	return err
 }
 
+func (b *Branch) LoadRepo(ctx context.Context) (err error) {
+	if b.Repo == nil && b.RepoID > 0 {
+		b.Repo, err = repo_model.GetRepositoryByID(ctx, b.RepoID)
+		if repo_model.IsErrRepoNotExist(err) {
```

Comment:
Why ignore this error? I think a branch should always have related branch.

Reply:
Done in https://github.com/go-gitea/gitea/pull/25812/commits/7fa0362d3637537d7c6fbc1064af0a0c2acc6b54